apps/src/templates/RegionalPartnerSearch.jsx

```diff
@@ -40,23 +41,63 @@ class RegionalPartnerSearch extends Component {
     responsiveSize: PropTypes.oneOf(['lg', 'md', 'sm', 'xs']).isRequired
   };
 
-  state = {
-    partnerInfo: undefined,
-    stateValue: "",
-    zipValue: "",
-    error: false,
-    loading: false
+  constructor(props) {
+    super(props);
+
+    let showZip = true;
+    let error = false;
+    let loading = false;
+
+    const partnerId = queryString.parse(window.location.search).partner;
+
+    if (partnerId) {
+      if (partnerId === "0") {
+        showZip = false;
+        error = WorkshopSearchErrors.no_partner;
+      } else {
+        $.ajax({
+          url: "/dashboardapi/v1/regional_partners/show/" + partnerId,
+          type: "get",
+          dataType: "json",
+          jsonp: false
+        }).done(this.partnerIdSuccess).fail(this.partnerIdFail);
+
+        showZip = false;
+        loading = true;
+      }
+    }
+
+    this.state = {
+      showZip: showZip,
+      partnerInfo: undefined,
+      stateValue: "",
+      zipValue: "",
+      error: error,
+      loading: loading
+    };
+  }
+
+  partnerIdSuccess = (response) => {
+    if (response.error) {
+      this.setState({showZip: true, error: false, loading: false});
+    } else {
+      this.setState({partnerInfo: response, loading: false});
+    }
```

Comment:
I'm confused by this. If `response.error` (is true), then we set the `state.error` to false?

Reply:
The idea is that if the partner ID provided in the URL doesn't match a partner, we just reset the control to its normal mode, showing the ZIP code input.  But good catch, because `state.error` had never been set, so resetting it wasn't necessary here.